src/battle/action_cmd/power_shock.c

```diff
@@ -2,12 +2,321 @@
 
 #define NAMESPACE action_command_power_shock
 
-INCLUDE_ASM(s32, "battle/action_cmd/power_shock", func_802A9000_42CF10);
+extern s32 D_80108AFC;
+extern s32 D_80108B28;
+extern s32 D_80108B80;
+extern s32 D_80109244;
+extern s32 D_80292974;
+extern s32 D_802943A0;
+extern s16 D_802A9AA2_42D9B2[];
+extern s32 D_802A9B00;
 
-INCLUDE_ASM(s32, "battle/action_cmd/power_shock", func_802A9210_42D120);
+ApiStatus N(CreateHudElements)(Evt* script, s32 isInitialCall) {
+    ActionCommandStatus* actionCommandStatus = &gActionCommandStatus;
+    BattleStatus* battleStatus = &gBattleStatus;
+    HudElement* hudElement;
+    s32 temp_v1;
 
+    battleStatus->unk_82 = 0x64;
+    battleStatus->unk_434 = &D_802943A0;
+
+    if (battleStatus->unk_83 == 0) {
+        battleStatus->actionSuccess = 0;
+        return ApiStatus_DONE2;
```

Comment:
If this return is needed, the rest of the function shouldn't be in an else block and should just be after this if

Reply:
Update; might be worth changing tidal_wave's (and others'?) as well.

Reply:
oh..if those are how this was then yeah that's not ideal

Reply:
Yeah it looks like this check is for a kind of universal "fail all action commands?" bool